controller/scheduler/swarm.py

```diff
@@ -1,21 +1,22 @@
 import re
 import time
-
+import logging
 from django.conf import settings
+from docker import errors
 from docker import Client
-
+from api.models import log_event
 from .states import JobState
 
-
 MATCH = re.compile(
     r'(?P<app>[a-z0-9-]+)_?(?P<version>v[0-9]+)?\.?(?P<c_type>[a-z-_]+)?.(?P<c_num>[0-9]+)')
 
 
 class SwarmClient(object):
 
-    def __init__(self, target, auth, options, pkey):
+    def __init__(self, target, auth, options, pkey, app):
```

Comment:
I agree that this is supposed to be a quick fix, but there has to be a middle-ground between quick fixes (adding tech debt) and clean fixes (potentially removing tech debt). This release has been mostly about removing tech debt. @mboersma has the same concern with adding this to the signature with his comment at https://github.com/deis/deis/pull/3692#issuecomment-104003783, so it sounds like this is something that may need to be refactored. I'd be err'ing on the side of getting this done right rather than getting this done fast, considering that this is a QoL change.